src/main.cpp

```diff
@@ -0,0 +1,111 @@
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <rerun.hpp>
+
+template <>
+struct rerun::ComponentBatchAdapter<rerun::components::Position3D, std::vector<Eigen::Vector3f>> {
+    // Sanity check that this is binary compatible.
+    static_assert(sizeof(components::Position3D) == sizeof(Eigen::Vector3f));
+    static_assert(alignof(components::Position3D) <= alignof(Eigen::Vector3f));
```

Comment:
ComponentBatch should have `borrow_compatible` which does these and a few other checks checks for you on 💡